focus-ios/Blockzilla/Extensions/StringExtensions.swift

```diff
@@ -23,12 +23,16 @@ fileprivate extension String {
 
 extension String {
     var isUrl: Bool {
-        let detector = try! NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue)
-        guard let match = detector.firstMatch(in: self, range: NSRange(location: 0, length: self.count)), match.range.length == self.count else {
-            return false
-        }
+        do {
+            let detector = try NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue)
+            guard let match = detector.firstMatch(in: self, range: NSRange(location: 0, length: self.count)), match.range.length == self.count else {
+                return false
+            }
 
-        return true
+            return true
+        } catch {
+            fatalError("Couldn't find data detector for type link")
```

Comment:
Minor nit: we could probably avoid the `fatalError` here, and add an assertion (which should crash in dev but not in release) and then return false as a default.

Reply:
hey, thanks for the suggestion. It makes sense as this would avoid a crash at least (if any). I have updated the PR please have a quick look. Thanks